src/xfer/mod.rs

```diff
@@ -0,0 +1,144 @@
+        parameters: &HashMap<String, String>,
+    ) -> Result<String> {
+        let content = self.get_from_cache_or_remote(url).await?;
+        templating::render(&content, parameters).context(TemplatingSnafu)
+    }
+
+    /// Retrieves data from the provided `url` and tries to deserialize it
+    /// into YAML.
+    pub async fn get_yaml_data<T>(&self, url: &Url) -> Result<T>
+    where
+        T: for<'a> Deserialize<'a> + Sized,
+    {
+        let content = self.get_from_cache_or_remote(url).await?;
+        serde_yaml::from_str(&content).context(YamlSnafu {})
+    }
+
+    /// Retrieves data from the provided `url`,  applies templating and tries
+    /// to deserialize it into YAML Variables inside handlebars are replaced
+    /// with values provided by the `parameters`, which is a key-value map.
+    pub async fn get_templated_yaml_data<T>(
+        &self,
+        url: &Url,
+        parameters: &HashMap<String, String>,
+    ) -> Result<T>
+    where
+        T: for<'a> Deserialize<'a> + Sized,
+    {
+        let content = self.get_from_cache_or_remote(url).await?;
+        let content = templating::render(&content, parameters).context(TemplatingSnafu)?;
+        serde_yaml::from_str(&content).context(YamlSnafu {})
+    }
+
+    /// Internal method which either looks up the requested file in the cache
+    /// or retrieves it from the remote located at `url` when the cache missed
+    /// or is expired.
+    async fn get_from_cache_or_remote(&self, url: &Url) -> Result<String> {
+        let file_name = Self::extract_file_name(url)?;
+
+        match self
+            .cache
+            .retrieve(&file_name)
+            .await
+            .context(CacheSnafu {})?
+        {
+            CacheStatus::Hit(content) => Ok(content),
+            CacheStatus::Expired | CacheStatus::Miss => {
+                let content = self.get(url).await?;
+                self.cache
+                    .store(&file_name, &content)
+                    .await
+                    .context(CacheSnafu {})?;
+
+                Ok(content)
+            }
+        }
+    }
+
+    /// Internal call which executes a HTTP GET request to `url`.
+    async fn get(&self, url: &Url) -> Result<String> {
+        let req = self
+            .client
+            .get(url.clone())
+            .build()
+            .context(RequestSnafu {})?;
+        let result = self.client.execute(req).await.context(RequestSnafu {})?;
+
+        result.text().await.context(RequestSnafu {})
+    }
+
+    /// Returns the last URL path segment as the filename. It ensures that the
+    /// last segment contains at least one dot.
+    fn extract_file_name(url: &Url) -> Result<String> {
+        let segment = url
+            .path_segments()
+            .ok_or(TransferError::FileNameError)?
+            .last()
+            .ok_or(TransferError::FileNameError)?;
+
+        ensure!(segment.contains('.'), FileNameSnafu {});
+        Ok(segment.to_string())
+    }
```

Comment:
Why isn't this in the cache layer instead? It's only used for that, right?

Reply:
Removed in 37e7c2d